### ISSUE TITLE
Unsupported platform for getting memory: x86_64-darwin20

### ISSUE THREAD

**Issue by jeremybdk** [newrelic/newrelic-ruby-agent#474]
## Description
Hello,
when using macOS Big Sur, at least the latest beta, I'm encountering this issue when calling ```NewRelic::Agent::Samplers::MemorySampler.new.sampler``` during test and in the rails console
Here is the error message I'm getting: `NewRelic::Agent::Sampler::Unsupported: Unsupported platform for getting memory: x86_64-darwin20`

## Expected Behavior
I expect it to work using macOS Big Sur

## Steps to Reproduce
Install latest macOS Big Sur Beta
Install New Relic gem
Include gem in Rails Console
`NewRelic::Agent::Samplers::MemorySampler.new.sampler`
Error should be displayed

## Your Environment
Include as many relevant details about your environment as possible including the running version of the Ruby agent, the Ruby version being used and any other relevant environment information
newrelic_rpm (6.13.1)
macOS Big Sur 11.0  (20A5384c)

Thanks for the help,

**Comment by tannalynn** [comment]
Hello @jeremybdk! Thank you for bringing this to our attention. We haven’t yet added support for macOS Big Sur.
The command we’re currently using for mac >= 10.6 is
`ps -o rss `
Does this command work for you on Big Sur? If so, this shouldn’t be difficult for us to add support for this.

**Comment by jeremybdk** [comment]
Hello @tannalynn,
The command does work.
The issue seems to be related to the regex used to retrieve the platform.
I'm not really using NewRelic on my Mac, just coding on it, and now some of our code/tests are failing because of this.
That's how I noticed the error.

Thanks

**Comment by tannalynn** [comment]
@jeremybdk Great, thank you for checking that command. Yes, we will need to update the regex but we wanted to make sure that this command still worked for the newest version. 
Here is a pull request that will go into our next release that should fix the issue https://github.com/newrelic/newrelic-ruby-agent/pull/476
In the meantime however, you should be able to specify the branch in the gemfile like so
```
gem 'newrelic_rpm', 
  git: 'https://github.com/newrelic/newrelic-ruby-agent.git', 
  branch: 'update_platform_detection_big_sur'
```
I believe this should solve the issue for you, so if you could let me know that would be great.
Thanks!

**Comment by jeremybdk** [comment]
Perfect it's now working 👍 Thanks a lot for the fast fix!!!

**Comment by mwlang** [comment]
This fix will appear in the next release, 6.14.0
